project-api/ballerina-projects/src/main/java/io/ballerina/projects/balo/BaloFiles.java

```diff
@@ -0,0 +1,200 @@
+        tomlPackage.setVersion(packageJson.getVersion());
+        tomlPackage.setLicense(packageJson.getLicenses());
+        tomlPackage.setAuthors(packageJson.getAuthors());
+        tomlPackage.setRepository(packageJson.getSourceRepository());
+        tomlPackage.setKeywords(packageJson.getKeywords());
+
+        ballerinaToml.setPkg(tomlPackage);
+        return ballerinaToml;
+    }
+
+    private static void validatePackageJson(PackageJson packageJson) {
+        if (packageJson.getOrganization() == null || "".equals(packageJson.getOrganization())) {
+            throw new RuntimeException("'organization' does not exists in 'package.json'");
+        }
+        if (packageJson.getName() == null || "".equals(packageJson.getName())) {
+            throw new RuntimeException("'name' does not exists in 'package.json'");
+        }
+        if (packageJson.getVersion() == null || "".equals(packageJson.getVersion())) {
+            throw new RuntimeException("'version' does not exists in 'package.json'");
+        }
+    }
+
+    private static ModuleData loadModule(Path modulePath) {
+        // check module path exists
+        if (!Files.exists(modulePath)) {
+            throw new RuntimeException("module does not exists:" + modulePath);
+        }
+
+        String moduleName = String.valueOf(modulePath.getFileName());
+        if (moduleName.contains(".")) { // not default module
+            moduleName = moduleName.split("\\.")[1];
+            moduleName = moduleName.replace("/", "");
+        }
+
+        // validate moduleName
+        if (!ProjectUtils.validateModuleName(moduleName)) {
+            throw new RuntimeException("Invalid module name : '" + moduleName + "' :\n" +
+                    "Module name can only contain alphanumerics, underscores and periods " +
+                    "and the maximum length is 256 characters");
+        }
+        List<DocumentData> srcDocs = loadDocuments(modulePath);
+        List<DocumentData> testSrcDocs = Collections.emptyList();
+
+        // TODO Read Module.md file. Do we need to? Balo creator may need to package Module.md
+        return ModuleData.from(modulePath, srcDocs, testSrcDocs);
+    }
+
+    private static List<DocumentData> loadDocuments(Path dirPath) {
+        try (Stream<Path> pathStream = Files.walk(dirPath, 1)) {
+            return pathStream
+                    .filter(matcher::matches)
+                    .map(BaloFiles::loadDocument)
+                    .collect(Collectors.toList());
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static DocumentData loadDocument(Path documentFilePath) {
+        Path fileNamePath = documentFilePath.getFileName();
+        // IMO, fileNamePath cannot be null in this case.
+        String name = fileNamePath != null ? fileNamePath.toString() : "";
+        return DocumentData.from(name, String.valueOf(documentFilePath));
```

Comment:
The 2nd argument passed to create an instance of DocumentData should be the content.

Reply:
Can we reuse the existing method in ProjectFiles here too?

Reply:
Added